### PR TITLE
More CRD docs!

### PR DESCRIPTION
**Pull request by fhennig**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] ~~CRD changes approved~~
- [x] Helm chart can be installed and deployed operator works
- [ ] ~~Integration tests passed (for non trivial changes)~~
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: rust/crd/src/authentication/mod.rs [hunk 1]
```diff
@@ -52,11 +52,7 @@ pub enum Error {
 #[derive(Clone, Deserialize, Debug, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct DruidAuthentication {
-    /// The AuthenticationClass <https://docs.stackable.tech/home/nightly/concepts/authenticationclass.html> to use.
-    ///
-    /// ## TLS provider
-    /// Please note that the SecretClass used to authenticate users needs to be the same
-    /// as the SecretClass used for internal communication.
+    /// The name of an [AuthenticationClass](DOCS_BASE_URL_PLACEHOLDER/concepts/authentication) object.
```
Comment (sbernauer) [inline rust/crd/src/authentication/mod.rs:55]: Why remove the documentation of the tls provider restriction? ;)

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -1016,13 +1060,16 @@ pub struct IngestionSpec {
...
     #[fragment_attrs(serde(default))]
     resources: Resources<storage::DruidStorage, NoRuntimeLimits>,
+    ///
     #[fragment_attrs(serde(default))]
     pub logging: Logging<Container>,
+    ///
     #[fragment_attrs(serde(default))]
     pub affinity: StackableAffinity,
-    /// Time period Pods have to gracefully shut down, e.g. `30m`, `1h` or `2d`. Consult the operator documentation for details.
+    /// TODO Time period Pods have to gracefully shut down, e.g. `30m`, `1h` or `2d`. Consult the operator documentation for details.
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:1072]: todo marker

Comment (fhennig) [reply rust/crd/src/lib.rs]: fixed: [055c6ba](https://github.com/stackabletech/druid-operator/pull/494/commits/055c6baa507da25dec9c2b6aefef2fdea8fae795)

File: rust/crd/src/authentication/mod.rs (hunk 1 above)
Comment (fhennig) [reply rust/crd/src/authentication/mod.rs:55]: https://github.com/stackabletech/druid-operator/pull/494/commits/8adfa6050d12e6de56e9c487f1360cd52bb6fca9

**Pull request merged**